File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_to_index = {}

        for i, num in enumerate(nums):
            x = target - num # Complement

            if x in num_to_index: # if complement already exists
                return [num_to_index[x], i] # return complement's index and current index(by doing so, there is no chance to use the same element twice)
            num_to_index[num] = i # otherwise, store the nums in the dictionary first.

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] answer = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                answer[0] = map.get(complement);
                answer[1] = i;
            }
            map.put(nums[i], i);
        }
        return answer;
    }
}